airflow/providers/openlineage/utils/utils.py

```diff
@@ -0,0 +1,482 @@
+    file_name = os.path.basename(file_path)
+    cwd = os.path.dirname(abs_path)
+
+    # get the repo url
+    repo_url = execute_git(cwd, ["config", "--get", "remote.origin.url"])
+
+    # get the repo relative path
+    repo_relative_path = execute_git(cwd, ["rev-parse", "--show-prefix"])
+
+    # get the commitId for the particular file
+    commit_id = execute_git(cwd, ["rev-list", "HEAD", "-1", "--", file_name])
+
+    # build the URL
+    base_url = url_to_https(repo_url)
+    if not base_url:
+        return None
+
+    return f"{base_url}/blob/{commit_id}/{repo_relative_path}{file_name}"
+
+
+def get_task_location(task):
+    try:
+        if hasattr(task, "file_path") and task.file_path:
+            return get_location(task.file_path)
+        else:
+            return get_location(task.dag.fileloc)
+    except Exception:
+        return None
+
+
+def execute_git(cwd, params):
+    p = subprocess.Popen(["git"] + params, cwd=cwd, stdout=subprocess.PIPE, stderr=None)
+    p.wait(timeout=0.5)
+    out, err = p.communicate()
+    return out.decode("utf8").strip()
+
+
+def redacted_connection_uri(conn: Connection, filtered_params=None, filtered_prefixes=None):
+    """
+    Return the connection URI for the given Connection.
+    This method additionally filters URI by removing query parameters that are known to carry sensitive data
+    like username, password, access key.
+    """
+    if filtered_prefixes is None:
+        filtered_prefixes = []
+    if filtered_params is None:
+        filtered_params = []
+
+    def filter_key_params(k: str):
+        return k not in filtered_params and any(substr in k for substr in filtered_prefixes)
+
+    conn_uri = conn.get_uri()
+    parsed = urlparse(conn_uri)
+
+    # Remove username and password
+    netloc = f"{parsed.hostname}" + (f":{parsed.port}" if parsed.port else "")
+    parsed = parsed._replace(netloc=netloc)
+    if parsed.query:
+        query_dict = dict(parse_qsl(parsed.query))
+        if conn.EXTRA_KEY in query_dict:
+            query_dict = json.loads(query_dict[conn.EXTRA_KEY])
+        filtered_qs = {k: v for k, v in query_dict.items() if not filter_key_params(k)}
+        parsed = parsed._replace(query=urlencode(filtered_qs))
+    return urlunparse(parsed)
+
+
+def get_normalized_postgres_connection_uri(conn):
+    """
+    URIs starting with postgresql:// and postgres:// are both valid
+    PostgreSQL connection strings. This function normalizes it to
+    postgres:// as canonical name according to OpenLineage spec.
+    """
+    uri = redacted_connection_uri(conn)
+    if uri.startswith("postgresql"):
+        uri = uri.replace("postgresql", "postgres", 1)
+    return uri
+
+
+def get_connection(conn_id) -> Connection | None:
+    from airflow.hooks.base import BaseHook
+
+    with suppress(Exception):
+        return BaseHook.get_connection(conn_id=conn_id)
+    return None
+
+
+def get_job_name(task):
+    return f"{task.dag_id}.{task.task_id}"
```

Comment:
We model mapped tasks as the same job - with facet providing additional info. https://github.com/apache/airflow/pull/29940/files#diff-9160b0ef03eafca04d898afdebfdbfba285f09ba7f37bb70a8b6cf592905078cR68

This could be expanded, of course. We generally think it's better to emit more events and make lineage backends handle them, rather than limiting it.